spec/services/site_setting/update_spec.rb

```diff
@@ -84,5 +85,11 @@
         end
       end
     end
+
+    context "when one setting is having invalid value" do
```

Comment:
Can we extend this further and add a test case for when multiple settings are updated and one of them fails and ensure that none of the settings are actually updated.